pkg/cmd/openshift-controller-manager/controller/image.go

```diff
@@ -63,51 +58,41 @@ func (c *ImageTriggerControllerConfig) RunController(ctx ControllerContext) (boo
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
-	if !c.HasStatefulSetsEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "apps", Resource: "statefulsets"},
-			Informer:  ctx.ExternalKubeInformers.Apps().V1beta1().StatefulSets().Informer(),
-			Store:     ctx.ExternalKubeInformers.Apps().V1beta1().StatefulSets().Informer().GetIndexer(),
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
-	if !c.HasCronJobsEnabled {
-		sources = append(sources, imagetriggercontroller.TriggerSource{
-			Resource:  schema.GroupResource{Group: "batch", Resource: "cronjobs"},
-			Informer:  ctx.ExternalKubeInformers.Batch().V2alpha1().CronJobs().Informer(),
-			Store:     ctx.ExternalKubeInformers.Batch().V2alpha1().CronJobs().Informer().GetIndexer(),
-			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
-			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
-		})
-	}
+	sources = append(sources, imagetriggercontroller.TriggerSource{
+		Resource:  schema.GroupResource{Group: "build.openshift.io", Resource: "buildconfigs"},
+		Informer:  ctx.BuildInformers.Build().InternalVersion().BuildConfigs().Informer(),
+		Store:     ctx.BuildInformers.Build().InternalVersion().BuildConfigs().Informer().GetIndexer(),
+		TriggerFn: triggerbuildconfigs.NewBuildConfigTriggerIndexer,
+		Reactor:   triggerbuildconfigs.NewBuildConfigReactor(bcInstantiator, kclient.Core().RESTClient()),
+	})
+	sources = append(sources, imagetriggercontroller.TriggerSource{
+		Resource:  schema.GroupResource{Group: "extensions", Resource: "deployments"},
+		Informer:  ctx.ExternalKubeInformers.Extensions().V1beta1().Deployments().Informer(),
+		Store:     ctx.ExternalKubeInformers.Extensions().V1beta1().Deployments().Informer().GetIndexer(),
+		TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+		Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
+	})
+	sources = append(sources, imagetriggercontroller.TriggerSource{
+		Resource:  schema.GroupResource{Group: "extensions", Resource: "daemonsets"},
+		Informer:  ctx.ExternalKubeInformers.Extensions().V1beta1().DaemonSets().Informer(),
+		Store:     ctx.ExternalKubeInformers.Extensions().V1beta1().DaemonSets().Informer().GetIndexer(),
+		TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+		Reactor:   &triggerannotations.AnnotationReactor{Updater: updater},
+	})
+	sources = append(sources, imagetriggercontroller.TriggerSource{
+		Resource:  schema.GroupResource{Group: "apps", Resource: "statefulsets"},
+		Informer:  ctx.ExternalKubeInformers.Apps().V1beta1().StatefulSets().Informer(),
+		Store:     ctx.ExternalKubeInformers.Apps().V1beta1().StatefulSets().Informer().GetIndexer(),
```

Comment:
While you're at it can you update all these to use the `Apps().V1()` for all of the above 3 (deployments, daemonsets and statefulsets).

Reply:
> While you're at it can you update all these to use the Apps().V1() for all of the above 3 (deployments, daemonsets and statefulsets).

Doesn't that ripple through the types in the controller too?  I'd have to update all the trigger funcs, rigth?

Reply:
Ok, I'll create a followup.